pkgs/development/tools/language-servers/jdt-language-server/default.nix

```diff
@@ -1,102 +1,56 @@
 { lib
 , stdenv
-, fetchurl
-, makeWrapper
-, jdk
+, fetchzip
+, python3
 }:
 
-stdenv.mkDerivation rec {
+let
+  timestamp = "202312071447";
+in
+stdenv.mkDerivation (finalAttrs: {
   pname = "jdt-language-server";
-  version = "1.26.0";
-  timestamp = "202307271613";
+  version = "1.30.1";
 
-  src = fetchurl {
-    url = "https://download.eclipse.org/jdtls/milestones/${version}/jdt-language-server-${version}-${timestamp}.tar.gz";
-    sha256 = "sha256-ul/l7jsqg5UofiSu8gzm4Xg0z46HcRfmyqysamiKbFM=";
+  src = fetchzip {
```

Comment:
Could you just explain the switch from `fetchurl` to `fetchzip`? I am just not familiar with the distinction or why we should prefer this one

Reply:
fetchurl will fetch whatever file and leave it as is, it will be $src of this derivation,  then it will be upto derivation builder to extract it

fetchzip will fetch file and extract it, therefore $src will be an extracted version, most likely a directory

Reply:
Is one better than the other in general? Tbh I didn't really have a reason for switching.

Reply:
the only thing I can think of is, with `fetchurl` you will put compressed src into store and only extract for build, with fetchzip you will put extracted src into store

Reply:
Looking at the existing code base, it seems like both `fetchzip` with `stripRoot = false` and `fetchurl` with `sourceRoot = "."` are used a lot. I think for this, I'll keep it as `fetchurl` to minimize changes in this PR since there doesn't seem like there are many differences.